apps/functions-subscription/src/adapters/azure/functions/__tests__/list-trials.test.ts

```diff
@@ -0,0 +1,124 @@
+import { describe, expect, it } from 'vitest';
+import * as TE from 'fp-ts/TaskEither';
+import { makeAValidListTrialRequest, supportHttpRequestHeaders } from './data';
+import { makeFunctionContext, makeTestSystemEnv } from './mocks';
+import { makeListTrialsHandler } from '../list-trials';
+import { aTrial } from '../../../../domain/__tests__/data';
+import { HttpRequest } from '@azure/functions';
+import { Trial, TrialId } from '../../../../domain/trial';
+import { NonEmptyString } from '@pagopa/ts-commons/lib/strings';
+
+describe('makeListTrialsHandler', () => {
+  const aSlimTrial = {
+    id: aTrial.id,
+    name: aTrial.name,
+    state: aTrial.state,
+    description: aTrial.description,
+  } as Trial;
+
+  const anotherSlimTrial = {
+    ...aSlimTrial,
+    id: 'anotherTrialId' as TrialId,
+    name: 'anotherTrialName' as NonEmptyString,
+  } as Trial;
+
+  it('should return 403 if x-user-groups header does not contain the correct group', async () => {
+    const request = new HttpRequest({
+      url: makeAValidListTrialRequest().url,
+      method: makeAValidListTrialRequest().method,
+      headers: {
+        ...supportHttpRequestHeaders,
+        'x-user-groups': 'Guest,AnotherGroup',
+      },
+    });
+    const env = makeTestSystemEnv();
+    const actual = await makeListTrialsHandler(env)(
+      request,
+      makeFunctionContext(),
+    );
+    expect(actual.status).toStrictEqual(403);
+    expect(await actual.json()).toMatchObject({
+      status: 403,
+    });
+  });
+
+  it('should return 500 when an error occurred', async () => {
+    const env = makeTestSystemEnv();
+
+    const error = new Error('Something went wrong');
+    env.listTrials.mockReturnValueOnce(TE.left(error));
+
+    const actual = await makeListTrialsHandler(env)(
+      makeAValidListTrialRequest(),
+      makeFunctionContext(),
+    );
+
+    expect(actual.status).toStrictEqual(500);
+    expect(await actual.json()).toMatchObject({
+      status: 500,
+      detail: error.message,
+    });
+  });
+
+  it('should return 200 when trials exists', async () => {
+    const env = makeTestSystemEnv();
+
+    env.listTrials.mockReturnValueOnce(
+      TE.right([aSlimTrial, anotherSlimTrial]),
+    );
```

Comment:
**issue**: following a previous comment, the `listTrials` function returns a list of `Trial`, but the name are different

I'd expect to return something like `aTrial, anotherTrial` of type `Trial` here, and them, when checking the returned value after the handler call, I'd check that the `items` property contains the slim trials.

This means that you can return both a `CREATING` and `CREATED` trials when you invoke the `env.listTrials` method, but the results should contain only the attributes defined into the OpenAPI spec.

Reply:
Updated in 3dc0bb86816f2a2f2f3fbcf3807f68fc3e45837a